internal/alertmanager/currentalert/process.go

```diff
@@ -305,24 +305,29 @@ func downScale(log logr.Logger, labels model.LabelSet, client client.Client) err
 			brokerId := strconv.Itoa(int(id))
 			keyVals = append(keyVals, brokerId, cr.Status.BrokersState[brokerId].GracefulActionState.CruiseControlState)
 		}
-
-		log.Info("downscale is skipped as there are brokers which are pending task to be initiated in CC or already have a running CC task", keyVals...)
-
+		log.Info("downscale is skipped as there are brokers which are pending task to be initiated in CC "+
+			"or already have a running CC task", keyVals...)
 		return nil
 	}
 
-	var brokerId string
+	var brokerID string
 	if broker, ok := labels["brokerId"]; ok {
-		brokerId = string(broker)
+		brokerID = string(broker)
 	} else {
-		cc := scale.NewCruiseControlScaler(string(labels["namespace"]), cr.Spec.GetKubernetesClusterDomain(), cr.Spec.CruiseControlConfig.CruiseControlEndpoint, cr.Name)
-		brokerId, err = cc.GetBrokerIDWithLeastPartition()
+		cruiseControlURL := scale.CruiseControlURLFromKafkaCluster(cr)
+		// FIXME: we should reuse the context of passed to AController.Start() here
+		cc, err := scale.NewCruiseControlScaler(context.TODO(), cruiseControlURL)
+		if err != nil {
+			log.Error(err, "failed to initialize Cruise Control Scaler", "cruise control url", cruiseControlURL)
```

Comment:
Since the error is passed back to callee no need to log the error `err` otherwise it may be logged twice (by the callee as well).

return errors.WrapIfWithDetails(err, "failed to initialize Cruise Control Scaler", "cruise control url", cruiseControlURL) should do the job

Reply:
Agreed. Fixed.